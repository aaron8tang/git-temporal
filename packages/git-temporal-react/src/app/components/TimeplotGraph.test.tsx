import React from 'react';
import { mount } from 'enzyme';

import { TimeplotGraph } from './TimeplotGraph';
import fiveCommits from 'testHelpers/mocks/fiveCommits';
import tenCommits from 'testHelpers/mocks/tenCommits';

describe('components/TimeplotGraph', () => {
  describe('when rendered with five commits, start and end dates)', () => {
    let wrapper;
    let renderSpy;

    beforeAll(() => {
      // @ts-ignore
      renderSpy = jest.spyOn(TimeplotGraph.prototype, 'renderTimeplotGraph');
      wrapper = mount(
        <TimeplotGraph
          commits={fiveCommits}
          startDate={fiveCommits[3].authorDate + 1}
          endDate={fiveCommits[2].authorDate - 1}
          earliestCommitDate={fiveCommits[0].authorDate}
          latestCommitDate={fiveCommits.slice(-1)[0].authorDate}
        />
      );
      wrapper.instance().focus();
    });

    beforeEach(() => {
      renderSpy.mockReset();
    });

    test('it should match snapshot (it should have rendered a block element for each line with <br><br>")', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('and then changing commits rendered should rerender with 10 commits', () => {
      wrapper.setProps({ commits: tenCommits });
      expect(wrapper).toMatchSnapshot();
      expect(renderSpy).toHaveBeenCalledTimes(1);
    });

    test('and then changing one of the ten commits, it should rerender', () => {
      // have to deep copy the array to avoid changing prev commits too
      const newCommits = tenCommits.map(commit => Object.assign({}, commit));
      newCommits[4].id = 'testId';
      wrapper.setProps({ commits: newCommits });
      expect(wrapper).toMatchSnapshot();
      expect(renderSpy).toHaveBeenCalledTimes(1);
    });
  });
  describe('when rendered with five commits, start and end dates, and callbacks)', () => {
    let wrapper;
    const onMouseEnterMock = jest.fn();
    const onMouseLeaveMock = jest.fn();
    const onMouseMoveMock = jest.fn();
    const onMouseDownMock = jest.fn();
    const onMouseUpMock = jest.fn();

    beforeAll(() => {
      wrapper = mount(
        <TimeplotGraph
          commits={fiveCommits}
          startDate={fiveCommits[3].authorDate + 1}
          endDate={fiveCommits[2].authorDate - 1}
          earliestCommitDate={fiveCommits[0].authorDate}
          latestCommitDate={fiveCommits.slice(-1)[0].authorDate}
          onMouseEnter={onMouseEnterMock}
          onMouseLeave={onMouseLeaveMock}
          onMouseMove={onMouseMoveMock}
          onMouseDown={onMouseDownMock}
          onMouseUp={onMouseUpMock}
        />
      );
      wrapper.instance().calibrateScales();
      wrapper.instance().focus();
    });

    test('MouseEnter should callback onMouseEnter', () => {
      wrapper.simulate('mouseenter');
      expect(onMouseEnterMock).toHaveBeenCalledTimes(1);
    });
    test('MouseLeave should callback onMouseLeave', () => {
      wrapper.simulate('mouseleave');
      expect(onMouseLeaveMock).toHaveBeenCalledTimes(1);
    });
    test('MouseMove should callback onMouseMove', () => {
      wrapper.simulate('mousemove');
      expect(onMouseMoveMock).toHaveBeenCalledTimes(1);
    });
    test('MouseDown should callback onMouseDown', () => {
      wrapper.simulate('mousedown');
      expect(onMouseDownMock).toHaveBeenCalledTimes(1);
    });
    test('MouseUp should callback onMouseUp', () => {
      wrapper.simulate('mouseup');
      expect(onMouseUpMock).toHaveBeenCalledTimes(1);
    });
  });
});
