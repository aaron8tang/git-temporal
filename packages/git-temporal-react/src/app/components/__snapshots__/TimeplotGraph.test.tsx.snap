// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TimeplotGraph when rendered with five commits, start and end dates) and then changing commits rendered should rerender with 10 commits 1`] = `
<TimeplotGraph
  commits={
    Array [
      Object {
        "authorDate": 1537542150,
        "authorName": "Bee Wilkerson",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 2,
            "linesDeleted": 74,
            "name": "README.md",
          },
        ],
        "hash": "abbf59e3a",
        "id": "abbf59e3af03cc872a52a8bb3a1dcfeb1b60d9bb",
        "linesAdded": 2,
        "linesDeleted": 74,
        "message": "Explain why this fork exists in README.md",
        "relativeDate": "2 days ago",
      },
      Object {
        "authorDate": 1537379312,
        "authorName": "Maksim Markelov",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/scheduler/src/TracingSubscriptions.js",
          },
        ],
        "hash": "e1a067dea",
        "id": "e1a067dea0ffcacd1f664f30cd14463b00f52fa7",
        "linesAdded": 1,
        "linesDeleted": 1,
        "message": "Fix circular dependency in TracingSubscriptions (#13689)",
        "relativeDate": "4 days ago",
      },
      Object {
        "authorDate": 1537359272,
        "authorName": "Heaven",
        "body": "* fix comment typo <br> <br>* Update Scheduler.js <br>",
        "files": Array [
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/scheduler/src/Scheduler.js",
          },
        ],
        "hash": "518812eeb",
        "id": "518812eeb82a524dd264be79ca0573ce39cde945",
        "linesAdded": 1,
        "linesDeleted": 1,
        "message": "Clarify comment (#13684)",
        "relativeDate": "4 days ago",
      },
      Object {
        "authorDate": 1537317391,
        "authorName": "Dan",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 0,
            "linesDeleted": 42,
            "name": "scripts/rollup/results.json",
          },
        ],
        "hash": "eeb817785",
        "id": "eeb817785c771362416fd87ea7d2a1a32dde9842",
        "linesAdded": 0,
        "linesDeleted": 42,
        "message": "Remove some old files from stats",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537316788,
        "authorName": "Dan Abramov",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 5,
            "linesDeleted": 4,
            "name": "fixtures/{schedule => scheduler}/index.html",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "fixtures/tracing/index.html",
          },
          Object {
            "linesAdded": 26,
            "linesDeleted": 26,
            "name": "fixtures/tracing/script.js",
          },
          Object {
            "linesAdded": 5,
            "linesDeleted": 2,
            "name": "fixtures/unstable-async/suspense/src/components/App.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "fixtures/unstable-async/suspense/src/index.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "fixtures/unstable-async/time-slicing/README.md",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "fixtures/unstable-async/time-slicing/src/index.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-art/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-art/src/ReactARTHostConfig.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-dom/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-dom/src/client/ReactDOMHostConfig.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-native-renderer/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-reconciler/package.json",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/react-reconciler/src/ReactFiberRoot.js",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/react-reconciler/src/ReactFiberScheduler.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-reconciler/src/__tests__/ReactTracing-test.internal.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-test-renderer/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react/package.json",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/react/src/ReactSharedInternals.js",
          },
          Object {
            "linesAdded": 5,
            "linesDeleted": 5,
            "name": "packages/react/src/__tests__/ReactProfiler-test.internal.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react/src/__tests__/ReactProfilerDevToolsIntegration-test.internal.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 7,
            "name": "packages/schedule/npm/index.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 7,
            "name": "packages/schedule/npm/tracing-profiling.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 7,
            "name": "packages/schedule/npm/tracing.js",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/{schedule => scheduler}/README.md",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/index.js",
          },
          Object {
            "linesAdded": 7,
            "linesDeleted": 0,
            "name": "packages/scheduler/npm/index.js",
          },
          Object {
            "linesAdded": 7,
            "linesDeleted": 0,
            "name": "packages/scheduler/npm/tracing-profiling.js",
          },
          Object {
            "linesAdded": 7,
            "linesDeleted": 0,
            "name": "packages/scheduler/npm/tracing.js",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/npm/umd/schedule-tracing.profiling.min.js => scheduler/npm/umd/scheduler-tracing.development.js}",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/npm/umd/schedule-tracing.production.min.js => scheduler/npm/umd/scheduler-tracing.production.min.js}",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/npm/umd/schedule-tracing.development.js => scheduler/npm/umd/scheduler-tracing.profiling.min.js}",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/npm/umd/schedule.production.min.js => scheduler/npm/umd/scheduler.development.js}",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/npm/umd/schedule.profiling.min.js => scheduler/npm/umd/scheduler.production.min.js}",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/npm/umd/schedule.development.js => scheduler/npm/umd/scheduler.profiling.min.js}",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/{schedule => scheduler}/package.json",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 0,
            "name": "packages/{schedule/src/Schedule.js => scheduler/src/Scheduler.js}",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 0,
            "name": "packages/{schedule => scheduler}/src/Tracing.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/TracingSubscriptions.js",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/src/__tests__/Schedule-test.internal.js => scheduler/src/__tests__/Scheduler-test.internal.js}",
          },
          Object {
            "linesAdded": 19,
            "linesDeleted": 19,
            "name": "packages/{schedule/src/__tests__/ScheduleDOM-test.js => scheduler/src/__tests__/SchedulerDOM-test.js}",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/src/__tests__/ScheduleUMDBundle-test.internal.js => scheduler/src/__tests__/SchedulerUMDBundle-test.internal.js}",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/__tests__/Tracing-test.internal.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/__tests__/Tracing-test.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/__tests__/TracingSubscriptions-test.internal.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 0,
            "name": "packages/{schedule => scheduler}/tracing.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/shared/forks/{Schedule.umd.js => Scheduler.umd.js}",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/shared/forks/{ScheduleTracing.umd.js => SchedulerTracing.umd.js}",
          },
          Object {
            "linesAdded": 6,
            "linesDeleted": 6,
            "name": "scripts/rollup/bundles.js",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "scripts/rollup/forks.js",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "scripts/rollup/modules.js",
          },
          Object {
            "linesAdded": 36,
            "linesDeleted": 36,
            "name": "scripts/rollup/results.json",
          },
        ],
        "hash": "7ea3ca1d1",
        "id": "7ea3ca1d13b1b609678fa1369f8a1020c3ecb976",
        "linesAdded": 213,
        "linesDeleted": 209,
        "message": "Rename schedule to scheduler (#13683)",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537299931,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 2,
            "linesDeleted": 1,
            "name": "CHANGELOG.md",
          },
        ],
        "hash": "9b7081664",
        "id": "9b70816642c68672da97e911d6d351fd3c596474",
        "linesAdded": 2,
        "linesDeleted": 1,
        "message": "Added another bullet to the CHANGELOG",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537299674,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "CHANGELOG.md",
          },
        ],
        "hash": "db9d51b65",
        "id": "db9d51b65caeec0cfd2b8e35df59bdaba3eaa31b",
        "linesAdded": 1,
        "linesDeleted": 1,
        "message": "Rename 'Schedule' header -> 'Schedule (Experimental)'",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537299572,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 12,
            "linesDeleted": 0,
            "name": "CHANGELOG.md",
          },
        ],
        "hash": "0823f845c",
        "id": "0823f845cf38141866e7e5164c80d75b4985df04",
        "linesAdded": 12,
        "linesDeleted": 0,
        "message": "16.5.2 CHANGELOG",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537295450,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 125,
            "linesDeleted": 69,
            "name": "scripts/rollup/results.json",
          },
        ],
        "hash": "bec2ddaf1",
        "id": "bec2ddaf157497ddca85e823e180536d8154a52f",
        "linesAdded": 125,
        "linesDeleted": 69,
        "message": "Update bundle sizes for 16.5.2 release",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537295450,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 2,
            "linesDeleted": 1,
            "name": "scripts/error-codes/codes.json",
          },
        ],
        "hash": "789e714bd",
        "id": "789e714bd738b0dd566f3cff33932b9586751fc5",
        "linesAdded": 2,
        "linesDeleted": 1,
        "message": "Update error codes for 16.5.2 release",
        "relativeDate": "5 days ago",
      },
    ]
  }
  endDate={1526296166}
  startDate={1537212013}
>
  <div
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "red",
          "height": 130,
          "left": 28.5268279049927,
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 3,
          "zIndex": 1,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "green",
          "height": 130,
          "left": 1634.1703409245522,
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 3,
          "zIndex": 1,
        }
      }
    />
  </div>
</TimeplotGraph>
`;

exports[`components/TimeplotGraph when rendered with five commits, start and end dates) and then changing one of the ten commits, it should rerender 1`] = `
<TimeplotGraph
  commits={
    Array [
      Object {
        "authorDate": 1537542150,
        "authorName": "Bee Wilkerson",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 2,
            "linesDeleted": 74,
            "name": "README.md",
          },
        ],
        "hash": "abbf59e3a",
        "id": "abbf59e3af03cc872a52a8bb3a1dcfeb1b60d9bb",
        "linesAdded": 2,
        "linesDeleted": 74,
        "message": "Explain why this fork exists in README.md",
        "relativeDate": "2 days ago",
      },
      Object {
        "authorDate": 1537379312,
        "authorName": "Maksim Markelov",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/scheduler/src/TracingSubscriptions.js",
          },
        ],
        "hash": "e1a067dea",
        "id": "e1a067dea0ffcacd1f664f30cd14463b00f52fa7",
        "linesAdded": 1,
        "linesDeleted": 1,
        "message": "Fix circular dependency in TracingSubscriptions (#13689)",
        "relativeDate": "4 days ago",
      },
      Object {
        "authorDate": 1537359272,
        "authorName": "Heaven",
        "body": "* fix comment typo <br> <br>* Update Scheduler.js <br>",
        "files": Array [
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/scheduler/src/Scheduler.js",
          },
        ],
        "hash": "518812eeb",
        "id": "518812eeb82a524dd264be79ca0573ce39cde945",
        "linesAdded": 1,
        "linesDeleted": 1,
        "message": "Clarify comment (#13684)",
        "relativeDate": "4 days ago",
      },
      Object {
        "authorDate": 1537317391,
        "authorName": "Dan",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 0,
            "linesDeleted": 42,
            "name": "scripts/rollup/results.json",
          },
        ],
        "hash": "eeb817785",
        "id": "eeb817785c771362416fd87ea7d2a1a32dde9842",
        "linesAdded": 0,
        "linesDeleted": 42,
        "message": "Remove some old files from stats",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537316788,
        "authorName": "Dan Abramov",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 5,
            "linesDeleted": 4,
            "name": "fixtures/{schedule => scheduler}/index.html",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "fixtures/tracing/index.html",
          },
          Object {
            "linesAdded": 26,
            "linesDeleted": 26,
            "name": "fixtures/tracing/script.js",
          },
          Object {
            "linesAdded": 5,
            "linesDeleted": 2,
            "name": "fixtures/unstable-async/suspense/src/components/App.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "fixtures/unstable-async/suspense/src/index.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "fixtures/unstable-async/time-slicing/README.md",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "fixtures/unstable-async/time-slicing/src/index.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-art/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-art/src/ReactARTHostConfig.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-dom/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-dom/src/client/ReactDOMHostConfig.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-native-renderer/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-reconciler/package.json",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/react-reconciler/src/ReactFiberRoot.js",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/react-reconciler/src/ReactFiberScheduler.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-reconciler/src/__tests__/ReactTracing-test.internal.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react-test-renderer/package.json",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react/package.json",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/react/src/ReactSharedInternals.js",
          },
          Object {
            "linesAdded": 5,
            "linesDeleted": 5,
            "name": "packages/react/src/__tests__/ReactProfiler-test.internal.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/react/src/__tests__/ReactProfilerDevToolsIntegration-test.internal.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 7,
            "name": "packages/schedule/npm/index.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 7,
            "name": "packages/schedule/npm/tracing-profiling.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 7,
            "name": "packages/schedule/npm/tracing.js",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/{schedule => scheduler}/README.md",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/index.js",
          },
          Object {
            "linesAdded": 7,
            "linesDeleted": 0,
            "name": "packages/scheduler/npm/index.js",
          },
          Object {
            "linesAdded": 7,
            "linesDeleted": 0,
            "name": "packages/scheduler/npm/tracing-profiling.js",
          },
          Object {
            "linesAdded": 7,
            "linesDeleted": 0,
            "name": "packages/scheduler/npm/tracing.js",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/npm/umd/schedule-tracing.profiling.min.js => scheduler/npm/umd/scheduler-tracing.development.js}",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/npm/umd/schedule-tracing.production.min.js => scheduler/npm/umd/scheduler-tracing.production.min.js}",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/npm/umd/schedule-tracing.development.js => scheduler/npm/umd/scheduler-tracing.profiling.min.js}",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/npm/umd/schedule.production.min.js => scheduler/npm/umd/scheduler.development.js}",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/npm/umd/schedule.profiling.min.js => scheduler/npm/umd/scheduler.production.min.js}",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/npm/umd/schedule.development.js => scheduler/npm/umd/scheduler.profiling.min.js}",
          },
          Object {
            "linesAdded": 2,
            "linesDeleted": 2,
            "name": "packages/{schedule => scheduler}/package.json",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 0,
            "name": "packages/{schedule/src/Schedule.js => scheduler/src/Scheduler.js}",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 0,
            "name": "packages/{schedule => scheduler}/src/Tracing.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/TracingSubscriptions.js",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "packages/{schedule/src/__tests__/Schedule-test.internal.js => scheduler/src/__tests__/Scheduler-test.internal.js}",
          },
          Object {
            "linesAdded": 19,
            "linesDeleted": 19,
            "name": "packages/{schedule/src/__tests__/ScheduleDOM-test.js => scheduler/src/__tests__/SchedulerDOM-test.js}",
          },
          Object {
            "linesAdded": 8,
            "linesDeleted": 8,
            "name": "packages/{schedule/src/__tests__/ScheduleUMDBundle-test.internal.js => scheduler/src/__tests__/SchedulerUMDBundle-test.internal.js}",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/__tests__/Tracing-test.internal.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/__tests__/Tracing-test.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/{schedule => scheduler}/src/__tests__/TracingSubscriptions-test.internal.js",
          },
          Object {
            "linesAdded": 0,
            "linesDeleted": 0,
            "name": "packages/{schedule => scheduler}/tracing.js",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/shared/forks/{Schedule.umd.js => Scheduler.umd.js}",
          },
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "packages/shared/forks/{ScheduleTracing.umd.js => SchedulerTracing.umd.js}",
          },
          Object {
            "linesAdded": 6,
            "linesDeleted": 6,
            "name": "scripts/rollup/bundles.js",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "scripts/rollup/forks.js",
          },
          Object {
            "linesAdded": 4,
            "linesDeleted": 4,
            "name": "scripts/rollup/modules.js",
          },
          Object {
            "linesAdded": 36,
            "linesDeleted": 36,
            "name": "scripts/rollup/results.json",
          },
        ],
        "hash": "7ea3ca1d1",
        "id": "testId",
        "linesAdded": 213,
        "linesDeleted": 209,
        "message": "Rename schedule to scheduler (#13683)",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537299931,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 2,
            "linesDeleted": 1,
            "name": "CHANGELOG.md",
          },
        ],
        "hash": "9b7081664",
        "id": "9b70816642c68672da97e911d6d351fd3c596474",
        "linesAdded": 2,
        "linesDeleted": 1,
        "message": "Added another bullet to the CHANGELOG",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537299674,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 1,
            "linesDeleted": 1,
            "name": "CHANGELOG.md",
          },
        ],
        "hash": "db9d51b65",
        "id": "db9d51b65caeec0cfd2b8e35df59bdaba3eaa31b",
        "linesAdded": 1,
        "linesDeleted": 1,
        "message": "Rename 'Schedule' header -> 'Schedule (Experimental)'",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537299572,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 12,
            "linesDeleted": 0,
            "name": "CHANGELOG.md",
          },
        ],
        "hash": "0823f845c",
        "id": "0823f845cf38141866e7e5164c80d75b4985df04",
        "linesAdded": 12,
        "linesDeleted": 0,
        "message": "16.5.2 CHANGELOG",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537295450,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 125,
            "linesDeleted": 69,
            "name": "scripts/rollup/results.json",
          },
        ],
        "hash": "bec2ddaf1",
        "id": "bec2ddaf157497ddca85e823e180536d8154a52f",
        "linesAdded": 125,
        "linesDeleted": 69,
        "message": "Update bundle sizes for 16.5.2 release",
        "relativeDate": "5 days ago",
      },
      Object {
        "authorDate": 1537295450,
        "authorName": "Brian Vaughn",
        "body": "",
        "files": Array [
          Object {
            "linesAdded": 2,
            "linesDeleted": 1,
            "name": "scripts/error-codes/codes.json",
          },
        ],
        "hash": "789e714bd",
        "id": "789e714bd738b0dd566f3cff33932b9586751fc5",
        "linesAdded": 2,
        "linesDeleted": 1,
        "message": "Update error codes for 16.5.2 release",
        "relativeDate": "5 days ago",
      },
    ]
  }
  endDate={1526296166}
  startDate={1537212013}
>
  <div
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
    style={
      Object {
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "red",
          "height": 130,
          "left": 28.5268279049927,
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 3,
          "zIndex": 1,
        }
      }
    />
    <div
      style={
        Object {
          "backgroundColor": "green",
          "height": 130,
          "left": 1634.1703409245522,
          "opacity": 0.5,
          "position": "absolute",
          "top": 0,
          "width": 3,
          "zIndex": 1,
        }
      }
    />
  </div>
</TimeplotGraph>
`;

exports[`components/TimeplotGraph when rendered with five commits, start and end dates) it should match snapshot (it should have rendered a block element for each line with <br><br>") 1`] = `undefined`;
